### ISSUE TITLE
Calling UI Only functions on background thread causes crash

### ISSUE THREAD

**Issue by mweyamutsvene** [Instabug/instabug-cordova#90]
Usually when you just launch the app the system seems to ignore these calls, but if you are running UI tests in xcode, ui calls, that are meant to be run on the main thread, often crash the app making testing on an actual device and newer versions of iOS SDK impossible.  

```
=================================================================
Main Thread Checker: UI API called on a background thread: -[UIApplication setNetworkActivityIndicatorVisible:]
PID: 51759, TID: 17228708, Thread name: (none), Queue name: NSOperationQueue 0x14bf05990 (QOS: UNSPECIFIED), QoS: 0
Backtrace:
4   Organizer                           0x00000001047adc84 -[IBGServerManager updateNetworkActivityIndicator] + 144
5   Organizer                           0x00000001047adbe0 -[IBGServerManager refresh] + 292
6   Organizer                           0x00000001047add58 -[IBGServerManager handleRequest:] + 52
7   Organizer                           0x00000001047dca00 -[IBGFeatures featuresRequestDone:isSuccess:response:error:] + 864
8   Organizer                           0x00000001047bf4c8 -[IBGFeaturesRequest dataServerExecutionDone:statusCode:error:response:] + 272
9   Organizer                           0x000000010479f5f0 __67+[IBGServerDataRequest executeWithURL:HTTPMethod:payload:delegate:]_block_invoke_2 + 120
10  Organizer                           0x000000010479f1d0 -[IBGServerDataRequest prepareResponseWithData:error:rawResponse:completionHandler:] + 520
11  Organizer                           0x000000010479f534 __67+[IBGServerDataRequest executeWithURL:HTTPMethod:payload:delegate:]_block_invoke + 156
12  CFNetwork                           0x00000001ad0fedd0 _CFURLConnectionCopyTimingData + 18260
13  Foundation                          0x00000001aa1db410 503900AF-2ECD-329F-B742-C07C3E672BEB + 1131536
14  Foundation                          0x00000001aa0e5138 503900AF-2ECD-329F-B742-C07C3E672BEB + 123192
15  Foundation                          0x00000001aa1dd698 503900AF-2ECD-329F-B742-C07C3E672BEB + 1140376
16  Foundation                          0x00000001aa0e4dd0 503900AF-2ECD-329F-B742-C07C3E672BEB + 122320
17  Foundation                          0x00000001aa1de090 503900AF-2ECD-329F-B742-C07C3E672BEB + 1142928
18  Foundation                          0x00000001aa1ddb5c 503900AF-2ECD-329F-B742-C07C3E672BEB + 1141596
19  libdispatch.dylib                   0x00000001a9ac6998 A3849F96-1C9F-36C5-A15F-70C566F14CFF + 260504
20  libdispatch.dylib                   0x00000001a9ae3184 A3849F96-1C9F-36C5-A15F-70C566F14CFF + 377220
21  libdispatch.dylib                   0x00000001a9abce8c A3849F96-1C9F-36C5-A15F-70C566F14CFF + 220812
22  libdispatch.dylib                   0x00000001a9abc600 A3849F96-1C9F-36C5-A15F-70C566F14CFF + 218624
23  libdispatch.dylib                   0x00000001a9ac9110 A3849F96-1C9F-36C5-A15F-70C566F14CFF + 270608
24  libdispatch.dylib                   0x00000001a9ac98b0 A3849F96-1C9F-36C5-A15F-70C566F14CFF + 272560
25  libsystem_pthread.dylib             0x00000001a9b32b48 _pthread_wqthread + 212
26  libsystem_pthread.dylib             0x00000001a9b35760 start_wqthread + 8
```
You need to wrap these calls is a dispatch to the main thread.

**Comment by AliAbdelfattah** [comment]
@mweyamutsvene thanks for reporting this, it should be fixed in the upcoming release.

**Comment by alyezz** [comment]
@mweyamutsvene this issue has been fixed and is included in out latest release 9.1.6
